medusa/service/grpc/server.py

```diff
@@ -162,10 +167,14 @@ def configure_console_logging(config):
 server.add_insecure_port('[::]:50051')
 server.start()
 
+signal.signal(signal.SIGTERM, shutdown)
+
+server.wait_for_termination()
+
 # since server.start() will not block,
 # a sleep-loop is added to keep alive
-try:
-    while True:
-        time.sleep(86400)
-except KeyboardInterrupt:
-    server.stop(0)
+# try:
```

Comment:
Would it be possible to remove these lines if they're not used anymore?

Reply:
>There's a comment that looks like it could be removed and maybe it could be nice to have a small test somewhere that starts the server and then sends a SIGTERM to verify that the change works while protecting us from future regression.
wdyt?

Definitely. I will work on a test.

>Would it be possible to remove these lines if they're not used anymore?

👍

Reply:
With respect to testing, the issue is specific to containerized environments. If you run the gRPC server locally and send it `SIGTERM`, the python process will terminate. The integration tests already performa shutdown on the gRPC server, but they do not run in a containerized environment

Reply:
gotcha 👍